### PR TITLE
Verify script for multiarch test images

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:
This PR is to add verify script for checking images under `/test/images` are multiarch images
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```NONE
```

### CODE REVIEW

File: hack/.multiarch_images_exceptions.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+test/images/pets/redis
```
Comment (luxas) [inline hack/.multiarch_images_exceptions.txt]: Can you add `TODO(mkumatag)` here?

File: hack/make-rules/verify.sh [hunk 2]
```diff
@@ -38,6 +38,7 @@ QUICK_PATTERNS+=(
   "verify-boilerplate.sh"
   "verify-godep-licenses.sh"
   "verify-gofmt.sh"
+  "verify-multiarch-test-images.sh"
```
Comment (luxas) [inline hack/make-rules/verify.sh]: Can we rename this to just `verify-test-images.sh`?
I think that's more correct, as we can generally enforce the layout of that directory, not just that it's multiarch...

File: hack/verify-multiarch-test-images.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+cd "${KUBE_ROOT}"
+NON_MULTIARCH_IMAGES=()
+
+for IMAGE_DIR in $(find test/images -iname 'Dockerfile' -printf '%h\n'| LC_ALL=C sort -u)
+do
```
Comment (luxas) [inline hack/verify-multiarch-test-images.sh]: nit: `do` can be on the previous line

File: hack/verify-multiarch-test-images.sh [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    grep -q "^${IMAGE_DIR}$" hack/.multiarch_images_exceptions.txt && ret=$? || ret=$?
```
Comment (luxas) [inline hack/verify-multiarch-test-images.sh]: can't the two lines here be written as this?
Please comment about what it does as well...

`if [[ $(grep -q "^${IMAGE_DIR}$" hack/.multiarch_images_exceptions.txt >/dev/null; echo $?) != 0 ]]; then`

File: hack/verify-multiarch-test-images.sh [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    if [[ "${ret}" -ne 0 ]]; then
+        if [[ ! -f "${IMAGE_DIR}/VERSION" ]]; then
```
Comment (luxas) [inline hack/verify-multiarch-test-images.sh]: I'd like to enforce more rules here, but I think you should split this up into a separate func and just pass the image dir to validate.
e.g.
 - VERSION exists and is a `1.x` version
 - Dockerfile exists
 - something else?

File: hack/verify-multiarch-test-images.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+            NON_MULTIARCH_IMAGES+=("${IMAGE_DIR}")
+        fi
+    fi
+done
+
+if [ ${#NON_MULTIARCH_IMAGES[@]} -ne 0 ]; then
+    echo "[${NON_MULTIARCH_IMAGES[@]}] these images are not written for multiarchitecture."
```
Comment (luxas) [inline hack/verify-multiarch-test-images.sh]: Having the VERSION file there doesn't guarantee it's multiarch. I'd prefer to focus on the style/semantics/conventions for this dir instead.

**Review by luxas — COMMENTED** [review]
Thanks for creating this script, will be needed indeed!

File: hack/make-rules/verify.sh (hunk 2 above)
Comment (mkumatag) [inline hack/make-rules/verify.sh]: Already we have the script name by this name for checking the image names used in `test/e2e/*.go` files, not really sure whether we can place these in the same file!

File: hack/verify-multiarch-test-images.sh (hunk 6 above)
Comment (mkumatag) [inline hack/verify-multiarch-test-images.sh]: I don't see any specific style/semantic which applies for all the directories at the moment, so I kept only this check.

Comment (cblecker) [inline hack/verify-multiarch-test-images.sh]: Right now, this test is asserting that the existence of a VERSION file means the image is multiarch. As this isn't true, we need some better way of determining:
- if the image is multiarch
- that the multiarch satisfies the k8s definition.. is amd64 and arm good enough, or does it need arm64/ppc64le/s390x/etc?

**Comment by cblecker** [comment]
/unassign
Feel free to assign back when this is ready for review (assuming WIP due to self-assignment)

**Pull request closed without merging**